src/tree/subscriptionTreeItem.ts

```diff
@@ -72,20 +80,95 @@ class SubscriptionTreeItem extends AzExtParentTreeItem implements SubscriptionTr
         return false;
     }
 
-    public async loadMoreChildrenImpl(): Promise<AzExtTreeItem[]> {
-        const clusterResources = await getResources(
+    async fetchClustersAndFleets() {
+        // load child nodes
+        const clusterResourcesPromise = await getResources(
+            // get all clusters
             this.sessionProvider,
-            this.subscription.subscriptionId,
-            "Microsoft.ContainerService/managedClusters",
+            this.subscriptionId,
+            clusterResourceType,
         );
-        if (failed(clusterResources)) {
+        if (failed(clusterResourcesPromise)) {
             window.showErrorMessage(
-                `Failed to list clusters in subscription ${this.subscription.subscriptionId}: ${clusterResources.error}`,
+                `Failed to list clusters in subscription ${this.subscriptionId}: ${clusterResourcesPromise.error}`,
             );
-            return [];
+            throw clusterResourcesPromise.error;
+        }
+        const clusterResources = clusterResourcesPromise.result;
+
+        const fleetResourcesPromise = await getResources(
+            // get all fleets
+            this.sessionProvider,
+            this.subscriptionId,
+            fleetResourceType,
+        );
+        if (failed(fleetResourcesPromise)) {
+            window.showErrorMessage(
+                `Failed to list fleets in subscription ${this.subscriptionId}: ${fleetResourcesPromise.error}`,
+            );
+            throw fleetResourcesPromise.error;
         }
+        const fleetResources = fleetResourcesPromise.result;
```

Comment:
suggestion to simplify code. this const is unecessary : 
- delete L111 
- delete L97

change the return statement to:

`return { clusterResourcesPromise.result, fleetResourcesPromise.result };`